src/Illuminate/Support/Str.php

```diff
@@ -1736,6 +1736,18 @@ public static function uuid()
                     : Uuid::uuid4();
     }
 
+    /**
+     * Generate a UUID (version 7).
+     *
+     * @return \Ramsey\Uuid\UuidInterface
+     */
+    public static function uuid7()
+    {
+        return static::$uuidFactory
+                    ? call_user_func(static::$uuidFactory)
+                    : Uuid::uuid7();
+    }
```

Comment:
Is it really okay, to use the same uuidFactory for v4 and v7? Should it not be a different factory, in case you want to use both uuids at the same time?

Reply:
Sorry for taking shortcuts i was focused only on fixing the `HasV7Uuids`.
Now uuid7 is implemented fully like uuid4 and also it accepts optional time now like ulid please let me know if there is another changes that need to be added